src/middleware/web-outgoing.ts

```diff
@@ -34,7 +34,7 @@ const setRedirectHostRewrite = defineProxyOutgoingMiddleware(
       proxyRes.headers.location &&
       redirectRegex.test(String(proxyRes.statusCode))
     ) {
-      const target = new URL(options.target);
+      const target = options.target;
```

Comment:
Since I changed `Partial<URL>` to `URL`, this is no longer needed. (previously `Partial<URL> | URL | ProxyTargetDetailed` was passed, which probably doesn't work)